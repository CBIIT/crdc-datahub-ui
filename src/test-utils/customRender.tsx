import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import theme from "../theme/testTheme";

const AllTheProviders = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
