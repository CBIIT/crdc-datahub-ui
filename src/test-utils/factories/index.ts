export * from "./ApprovedStudyFactory";
export * from "./BatchFactory";
export * from "./SubmissionStatisticFactory";
export * from "./UserFactory";
