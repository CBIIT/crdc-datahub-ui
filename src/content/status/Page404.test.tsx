import { axe } from "vitest-axe";
import { BrowserRouter, RouterProvider, createMemoryRouter } from "react-router-dom";
import { FC } from "react";
import userEvent from "@testing-library/user-event";
import { render, waitFor } from "../../test-utils";
import Page from "./Page404";

const TestParent: FC = () => (
  <BrowserRouter>
    <Page />
  </BrowserRouter>
);

describe("404 Page Cases", () => {
  it("should not have any accessibility violations", async () => {
    const { container } = render(<TestParent />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it("should render statically and correctly", () => {
    const { container } = render(<TestParent />);

    expect(container).toMatchSnapshot();
  });

  it("should set the page title to 'Page Not Found'", async () => {
    render(<TestParent />);

    await waitFor(() => expect(document.title).toBe("Page Not Found"));
  });

  it("should navigate home when the 'Return Home' button is clicked", async () => {
    const router = createMemoryRouter(
      [
        {
          path: "/404",
          element: <Page />,
        },
        {
          path: "/",
          element: <div>home page</div>,
        },
      ],
      {
        initialEntries: ["/404"],
      }
    );

    const { getByText, getByRole } = render(<RouterProvider router={router} />);

    await waitFor(() => {
      expect(getByText(/page not found/i)).toBeInTheDocument();
    });

    userEvent.click(getByRole("button"));

    await waitFor(() => {
      expect(getByText(/home page/i)).toBeInTheDocument();
    });
  });
});
