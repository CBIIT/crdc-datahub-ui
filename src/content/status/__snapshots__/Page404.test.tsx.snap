// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`404 Page Cases > should render statically and correctly 1`] = `
<div>
  <div
    class="MuiStack-root css-1odcfns-MuiStack-root"
  >
    <div
      class="MuiBox-root css-1iatin6"
    >
      <img
        alt="404 icon"
        class="css-1g2o5dd"
        src="/src/assets/icons/404.png"
      />
      <h1
        class="MuiTypography-root MuiTypography-h1 css-oastk0-MuiTypography-root"
      >
        Page not found.
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 css-me1iuw-MuiTypography-root"
      >
        Looks like you got a little turned around.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-1uyj2ed-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Return Home
      </button>
    </div>
  </div>
</div>
`;
